app/src/main/java/org/eyeseetea/malariacare/data/remote/api/OkHttpClientDataSource.java

```diff
@@ -0,0 +1,119 @@
+package org.eyeseetea.malariacare.data.remote.api;
+
+import android.util.Log;
+
+import com.squareup.okhttp.Authenticator;
+import com.squareup.okhttp.Credentials;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.util.concurrent.TimeUnit;
+
+public class OkHttpClientDataSource {
+
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+
+    private static final String TAG = ".PullDhisApiDataSource";
+    private static org.eyeseetea.malariacare.domain.entity.Credentials mCredentials;
+
+    public OkHttpClientDataSource(org.eyeseetea.malariacare.domain.entity.Credentials credentials) {
+        mCredentials = credentials;
+    }
+
+    /**
+     * Call to DHIS Server
+     * @param url
+     * @param method
+     */
+    public static String executeCall(String url, String method)
+            throws Exception {
+        Response response = executeCall(null, url, method);
+        return response.body().string();
+    }
+
+    private static Response executeCall(JSONObject data, String url, String method) throws
```

Comment:
JSONObject data parameter has no sense because this method only is executed from the above method passing null value